moxy/org.eclipse.persistence.moxy/src/main/java/org/eclipse/persistence/jaxb/JAXBContext.java

```diff
@@ -1478,7 +1477,7 @@ private QName getSchemaTypeForTypeMappingInfo(Type type) {
                 if (name == null) {
                     Class<?> theClass = (Class) type;
                     //Change default for byte[] to Base64 (JAXB 2.0 default)
-                    if (type == CoreClassConstants.ABYTE || type == CoreClassConstants.APBYTE || type == Image.class || type == Source.class || theClass.getCanonicalName().equals("jakarta.activation.DataHandler")) {
+                    if (type == CoreClassConstants.ABYTE || type == CoreClassConstants.APBYTE || theClass.getCanonicalName().equals("java.awt.Image") || type == Source.class || theClass.getCanonicalName().equals("jakarta.activation.DataHandler")) {
```

Comment:
Possible NPE `getCanonicalName()` should return `null` in some cases, but I'm not sure if it happens in this context.

Reply:
* type will never be null, see also the next (pre-existing) part of the condition
* they are moxy specific, so oxm is not the right place for this; the consolidation of this sort of constants spread through `JAXBContext`, `MappingsGenerator`, `AnnotationsProcessor` and `SchemaGenerator` deserves separate PR/commit, I believe